src/main/software/amazon/event/ruler/input/WildcardParser.java

```diff
@@ -3,30 +3,33 @@
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
+import software.amazon.event.ruler.MatchType;
 
-import static software.amazon.event.ruler.input.Parser.ASTERISK_BYTE;
-import static software.amazon.event.ruler.input.Parser.BACKSLASH_BYTE;
+import static software.amazon.event.ruler.input.DefaultParser.ASTERISK_BYTE;
+import static software.amazon.event.ruler.input.DefaultParser.BACKSLASH_BYTE;
 
 /**
  * A parser to be used specifically for wildcard rules.
  */
-public class WildcardParser {
+@SuppressWarnings({"UseOfConcreteClass", "ObjectAllocationInLoop"})
```

Comment:
curious if you've considered suppressing individual methods first? Ideally you want to suppress the smallest part of the code (and add a comment on why suppression is necessary).

Reply:
My bad, I'll remove that.